c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -1377,6 +1381,22 @@ bool body_reader_then_handler(JSContext *cx, HandleObject body_owner, HandleValu
     fprintf(stderr, "Error: read operation on body ReadableStream didn't respond with a "
                     "Uint8Array. Received value: ");
     dump_value(cx, val, stderr);
+
+    // reject the request promise
+    if (Request::is_instance(body_owner)) {
+      RootedObject response_promise(cx, Request::response_promise(body_owner));
+      RootedValue exn(cx);
+
+      // TODO: this should be a TypeError, but I'm not sure how to make that work
+      JS_ReportErrorUTF8(cx, "TypeError");
+      if (!JS_GetPendingException(cx, &exn)) {
+        return false;
+      }
+      JS_ClearPendingException(cx);
+
+      return JS::RejectPromise(cx, response_promise, exn);
```

Comment:
@JakeChampion I'm not sure if this is the best way to construct the error, and the wpt test suggests that we should be raising a `TypeError` at this point. Do you happen to know the right error construction function to make a `TypeError`, and is it possible to construct an error without moving it out of the pending exception slot?

Reply:
To raise a different error class, the JS_ReportErrorNumber* functions need to be used (https://searchfox.org/mozilla-central/source/js/src/jsapi.cpp#3544-3627)

let's change this in another pull-request to typeerror as we have a separate pull-request which creates our own error definitions already -> https://github.com/fastly/js-compute-runtime/pull/164